geomesa-z3/src/main/scala/org/locationtech/geomesa/curve/Z3SFC.scala

```diff
@@ -55,10 +55,11 @@ class Z3SFC(period: TimePeriod, precision: Int = 21) extends SpaceTimeFillingCur
                       t: Seq[(Long, Long)],
                       precision: Int,
                       maxRanges: Option[Int]): Seq[IndexRange] = {
+    val maxRecursion = sys.props.get("geomesa.scan.ranges.recurse").map(_.toInt).orElse(Some(Int.MaxValue))
     val zbounds = for { (xmin, ymin, xmax, ymax) <- xy ; (tmin, tmax) <- t } yield {
       ZRange(index(xmin, ymin, tmin), index(xmax, ymax, tmax))
     }
-    Z3.zranges(zbounds.toArray, precision, maxRanges)
+    Z3.zranges(zbounds.toArray, precision, maxRanges, maxRecursion)
```

Comment:
Should we add this to Z2?  Maybe as a separate ticket with separate testing?

Reply:
yeah, i think z2 is less likely to suffer from this issue as the dimensions are better aligned. we could do it along with z2 seeking, but atm we're not really using z2 are we?

Reply:
The z2 index is in for spatial-only / context datasets.  This is subtle enough that a separate ticket (or waiting on it entirely) would be sensible to me!